tensorflowonspark/pipeline.py

```diff
@@ -0,0 +1,313 @@
+# Copyright 2017 Yahoo Inc.
+# Licensed under the terms of the Apache 2.0 license.
+# Please see LICENSE file in the project root for terms.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from pyspark.context import SparkContext
+from pyspark.ml.param.shared import *
+from pyspark.ml.pipeline import Estimator, Model, Pipeline
+from pyspark.sql import SparkSession
+
+import tensorflow as tf
+from tensorflow.contrib.saved_model.python.saved_model import reader, signature_def_utils
+from tensorflow.python.framework import ops as ops_lib
+from tensorflow.python.saved_model import loader
+from . import TFCluster, gpu_info
+
+import copy
+import logging
+import os
+import subprocess
+
+##### TensorFlowOnSpark Params
+
+class HasBatchSize(Params):
+  batch_size = Param(Params._dummy(), "batch_size", "Number of records per batch", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasBatchSize, self).__init__()
+  def setBatchSize(self, value):
+    return self._set(batch_size=value)
+  def getBatchSize(self):
+    return self.getOrDefault(self.batch_size)
+
+class HasClusterSize(Params):
+  cluster_size = Param(Params._dummy(), "cluster_size", "Number of nodes in the cluster", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasClusterSize, self).__init__()
+  def setClusterSize(self, value):
+    return self._set(cluster_size=value)
+  def getClusterSize(self):
+    return self.getOrDefault(self.cluster_size)
+
+class HasEpochs(Params):
+  epochs = Param(Params._dummy(), "epochs", "Number of epochs to train", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasEpochs, self).__init__()
+  def setEpochs(self, value):
+    return self._set(epochs=value)
+  def getEpochs(self):
+    return self.getOrDefault(self.epochs)
+
+class HasInputTensor(Params):
+  tensor_in = Param(Params._dummy(), "tensor_in", "Name of input tensor in signature def", typeConverter=TypeConverters.toString)
+  def __init__(self):
+    super(HasInputTensor, self).__init__()
+  def setInputTensor(self, value):
+    return self._set(tensor_in=value)
+  def getInputTensor(self):
+    return self.getOrDefault(self.tensor_in)
+
+class HasModelDir(Params):
+  model_dir = Param(Params._dummy(), "model_dir", "Path to save/load model checkpoints", typeConverter=TypeConverters.toString)
```

Comment:
Agreed.  Part of the challenge is that historically, Tensorflow "model" was where the model checkpoints were stored, where the "saved_model" was a subsequent layer added to make these model checkpoints more usable/servable.  I'll try to see if I can combine them, but otherwise, we can change the naming as needed.